### DISCUSSION TITLE
Latest version (2.10.3) still won't startup for me

### DISCUSSION THREAD

**Discussion by ervwalter** [NginxProxyManager/nginx-proxy-manager#2925] [General]
There were some similar issues, and some fixes in later updates, but the latest version still doesn't work for me.  Before I create a new issue, I thought I'd ask here what I can do to help debug, because the only thing that happens is that the startup gets stuck at:

```
❯ Configuring npm user ...
❯ Configuring npm group ...
❯ Checking paths ...
❯ Setting ownership ...
```

I don't know what debugging information would be helpful to understand what the container is doing while it is stuck there.

**Comment by jeremyatourville** [comment]
We need to see some logs to help determine what is happening.  Are you running docker-compose?  If so, run  `docker-compose up -d && docker-compose logs -f`  from the same location where your yaml file is located.

**Reply by ervwalter** [reply]
The "code"/grey text above was the entirely of the docker log.  It printed those 4 things and then hung on setting ownership until docker killed the container and tried again (presumably because it took too long to get healthy?).

I don't have that container anymore, so it's hard for me to do testing.  I essentially gave up and switched to traefik for that particular server.

I suspect that it may have had something to do with the filesystem type (cephfs) that was mounted into the container.  That was the only real difference between the VM where it wasn't starting and the VM where it still works fine (it works fine on my VPC simple local disk)

**Comment by lucaberta** [comment]
Not proud to revive an old thread, but it's the exact same issue I am facing, and I have also done the steps provided by @jeremyatourville by running the docker compose commands provided in the May 31st 2023 answer.

The logs show exactly the same content as seen in the original post, everything seem to be stuck at the "Setting ownership" phase.

I had already been running an homelab instance with success, using docker compose and a stock YAML file, but I wanted to move the stack into Portainer and also change the location of the /data and /letsencrypt directories elsewhere in my system, and also update the release of the code.

Ever since I tried all this, I have not been able to have working container run.

I have also tried to have homelab create the data and letsencrypt directories from scratch, which it did, but still it got stuck at the "Setting ownership" message.

Is there any way to troubleshoot deeper in the commands, to see what upsets things?

Portainer shows the same exact things, like it should, a running container and the same exact logs.

Any hints to better troubleshoot the issue?

Many thanks,

Luca

**Reply by lucaberta** [reply]
Disregard, it looks like the container unfroze, and a number of logs have now been created. Let's see what I can make out of it.

I will update the thread as required.

**Reply by lucaberta** [reply]
Quick heads up on the issue, it just took some 25 minutes for the container to start properly, so I was stuck on the "Setting ownership" phase for 25 minutes, then the logs continued with "Dynamic resolvers" and the rest of the startup phase.

I will keep tinkering...

**Reply by ton-An** [reply]
> Quick heads up on the issue, it just took some 25 minutes for the container to start properly, so I was stuck on the "Setting ownership" phase for 25 minutes, then the logs continued with "Dynamic resolvers" and the rest of the startup phase.
> 
> I will keep tinkering...

This is it!
